src/moscot/base/problems/problem.py

```diff
@@ -249,19 +250,22 @@ def __init__(
         self._a: Optional[ArrayLike] = None
         self._b: Optional[ArrayLike] = None
 
+        self._t: Optional[float] = None  # only needed for diffusion-based distances
```

Comment:
This feels like a design mistake, since it can be set via `set_graph_xy`. Is there an easy way to refactor this?

Reply:
I discussed this with @giovp, as I didn't find an easy solution.

One option would be to instantiate the cost matrix in `set_graph_xy` if `t is not None` (because then `t!=eps/4` and we need to instantiate the cost matrix). The limitation would be that in (F)GW problems, we need to instantiate the cost matrix anyways, which currently doesn't happen in `set_graph_xy` because we can also directly use the geodesic cost, e.g. via the `graph-construction` callback in `prepare`. 

I would also be fine with instantiating it in two different places, but @giovp you were against it, right

Reply:
so the problem is I think that there should only one place where the geodesic is instantitate, as a cost or as a geometry, and it should be in solve. Now I understand that the problem is that that step can be called by either the prepare (and hence instantiate the OTPRoblem) or the `set_graph_xy`. I am not sure what would be an easy solution for this, maybe we can think about behaviour rather than API as a guiding principle?

For example, my understanding is that, if the user wants to use the geodesic cost, it has:
- pass it to prepare and set t, in case of quadratic problem
- use set_graph_xy explicitly, in case of linear problem

Is that correct>?

Reply:
- in case of the quadratic problem, the user should still be able to use `set_graph_xy`
- in case of the linear problem, the user can also pass it via the `prepare` method.

The main difference is that
- if the user goes via the `prepare` method, we automatically set `4t=eps`
- if the user goes via the `set_graph_xy` method, `t` and `eps` are handled independently of each other